onnxruntime/core/providers/rocm/math/matmul.cc

```diff
@@ -0,0 +1,232 @@
+    //                                                       stride_A,
+    //                                                       &zero,
+    //                                                       reinterpret_cast<HipT*>(Y->template MutableData<T>()),
+    //                                                       ldc,
+    //                                                       stride_C,
+    //                                                       static_cast<int>(batch_count)));
+    ROCBLAS_RETURN_IF_ERROR(rocblasGemmStridedBatchedHelper(Base::RocblasHandle(),
+                                                          transB,
+                                                          transA,
+                                                          static_cast<int>(helper.N()),
+                                                          static_cast<int>(helper.M()),
+                                                          static_cast<int>(helper.K()),
+                                                          &alpha,
+                                                          reinterpret_cast<const HipT*>(right_X->template Data<T>()),
+                                                          ldb,
+                                                          stride_B,
+                                                          reinterpret_cast<const HipT*>(left_X->template Data<T>()),
+                                                          lda,
+                                                          stride_A,
+                                                          &zero,
+                                                          reinterpret_cast<HipT*>(Y->template MutableData<T>()),
+                                                          ldc,
+                                                          stride_C,
+                                                          static_cast<int>(batch_count)));
+    return Status::OK();
+  }
+
+  HipAsyncBuffer<const HipT*> left_arrays(this, helper.LeftOffsets().size());
+  HipAsyncBuffer<const HipT*> right_arrays(this, helper.RightOffsets().size());
+  HipAsyncBuffer<HipT*> output_arrays(this, helper.OutputOffsets().size());
+  MatMulComputeHelper::OffsetToArrays(reinterpret_cast<const HipT*>(left_X->template Data<T>()), helper.LeftOffsets(), left_arrays.CpuSpan());
+  MatMulComputeHelper::OffsetToArrays(reinterpret_cast<const HipT*>(right_X->template Data<T>()), helper.RightOffsets(), right_arrays.CpuSpan());
+  MatMulComputeHelper::OffsetToArrays(reinterpret_cast<HipT*>(Y->template MutableData<T>()), helper.OutputOffsets(), output_arrays.CpuSpan());
+  ORT_RETURN_IF_ERROR(left_arrays.CopyToGpu());
+  ORT_RETURN_IF_ERROR(right_arrays.CopyToGpu());
+  ORT_RETURN_IF_ERROR(output_arrays.CopyToGpu());
+
+  // note that onnxruntime OrtValue is row major, while hipblas is column major,
+  // so swap left/right operands
+  // HIPBLAS_RETURN_IF_ERROR(hipblasGemmBatchedHelper(
```

Comment:
Same, delete it?

Reply:
Plan to have a separate PR to replace all hipBLAS call with rocBLAS call.